spylib/utils/session_token.py

```diff
@@ -43,14 +44,27 @@ class SessionToken(BaseModel):
     jti: int
     sid: str
 
+    @root_validator()
+    def equal_iss_and_dest(cls, values: Dict[str, Any]):
+        domain = cls.__url_to_base(values.get('iss'))
+        try:
+            store_domain(domain)
+        except ValueError:
+            raise InvalidIssuerError(f"The domain {domain} is not a valid issuer.")
+        
+        if cls.__url_to_base(values.get('iss')) != cls.__url_to_base(values.get('dest')):
+            raise MismatchedHostError(
+                f'The issuer {cls.__url_to_base(values.get("iss"))} does not match '
+                f'the destination {cls.__url_to_base(values.get("dest"))}'
```

Comment:
Just a suggestion for readability for next time. No need to implement:
```suggestion
        iss_url = cls.__url_to_base(values.get('iss'))
        dest_url = cls.__url_to_base(values.get('dest'))
        if iss_url != dest_url:
            raise MismatchedHostError(
                f'The issuer {iss_url} does not match the destination {dest_url}'
```